File: sample-conference-videochat/sample-conference-videochat/Components/InfoScreen/InfoScreenViewController/InfoTableViewController.m
//
//  InfoTableViewController.m
//  sample-conference-videochat
//
//  Created by Vladimir Nybozhinsky on 12/30/18.
//  Copyright © 2018 Quickblox. All rights reserved.
//

#import "InfoTableViewController.h"
#import "InfoModel.h"
#import "InfoTableViewCell.h"
#import <Quickblox/QBASession.h>

NSString *const kInfoTableViewCellId = @"InfoTableViewCell";
NSString *const kLogoTableViewCellId = @"QBLogoTableViewCell";


@interface InfoTableViewController ()
{
    NSMutableArray<InfoModel*>* infoModels;
}

@end

@implementation InfoTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setupTableView];
}

- (void)setupTableView
{
    infoModels = [NSMutableArray new];
    
    NSString *appVersion = NSBundle.mainBundle.infoDictionary[@"CFBundleShortVersionString"];
    InfoModel *appVersionModel = [[InfoModel alloc] init];
    appVersionModel.title = @"Application version:";
    appVersionModel.info = [NSString stringWithFormat:@"%@", appVersion];
    [infoModels addObject:appVersionModel];
    
    InfoModel *quickBloxSdkVersionModel = [[InfoModel alloc] init];
    quickBloxSdkVersionModel.title = @"QuickBlox SDK version:";
    quickBloxSdkVersionModel.info = [NSString stringWithFormat:@"%@", QuickbloxFrameworkVersion];
    [infoModels addObject:quickBloxSdkVersionModel];
    
    InfoModel *wbrtcVersionModel = [[InfoModel alloc] init];
    wbrtcVersionModel.title = @"QuickbloxWebRTC version:";
    wbrtcVersionModel.info = [NSString stringWithFormat:@"%@", QuickbloxWebRTCFrameworkVersion];
    [infoModels addObject:wbrtcVersionModel];
    
    InfoModel *appIDModel = [[InfoModel alloc] init];
    appIDModel.title = @"Application ID:";
    appIDModel.info = [NSString stringWithFormat:@"%lu", (unsigned long)QBSettings.applicationID];
    [infoModels addObject:appIDModel];
    
    InfoModel *authKeyModel = [[InfoModel alloc] init];
    authKeyModel.title = @"Auhtorization key:";
    authKeyModel.info = [NSString stringWithFormat:@"%@", QBSettings.authKey];
    [infoModels addObject:authKeyModel];
    
    InfoModel *authSecretModel = [[InfoModel alloc] init];
    authSecretModel.title = @"Auhtorization secret:";
    authSecretModel.info = [NSString stringWithFormat:@"%@", QBSettings.authSecret];
    [infoModels addObject:authSecretModel];
    
    InfoModel *accountKeyModel = [[InfoModel alloc] init];
    accountKeyModel.title = @"Account key:";
    accountKeyModel.info = [NSString stringWithFormat:@"%@", QBSettings.accountKey];
    [infoModels addObject:accountKeyModel];
    
    InfoModel *apiDomainModel = [[InfoModel alloc] init];
    apiDomainModel.title = @"API domain:";
    apiDomainModel.info = [NSString stringWithFormat:@"%@", QBSettings.apiEndpoint];
    [infoModels addObject:apiDomainModel];
    
    InfoModel *chatDomainModel = [[InfoModel alloc] init];
    chatDomainModel.title = @"API domain:";
    chatDomainModel.info = [NSString stringWithFormat:@"%@", QBSettings.chatEndpoint];
    [infoModels addObject:chatDomainModel];
    
    InfoModel *janusServerURLModel = [[InfoModel alloc] init];
    janusServerURLModel.title = @"API domain:";
    janusServerURLModel.info = [NSString stringWithFormat:@"%@", QBRTCConfig.conferenceEndpoint];
    [infoModels addObject:janusServerURLModel];
    
    InfoModel *logoModel = [[InfoModel alloc] init];
    logoModel.title = @"logo";
    logoModel.info = @"logo";
    [infoModels addObject:logoModel];
    
    [self.tableView reloadData];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    if (infoModels.count > 0) {
        return infoModels.count;
    }
    return 0;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = UITableViewCell.new;
    if (indexPath.row != infoModels.count - 1) {
        InfoTableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:kInfoTableViewCellId];
        InfoModel *model = infoModels[indexPath.row];
        [cell applyInfo:model];
        return cell;
    } else if (indexPath.row == infoModels.count - 1) {
        UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:kLogoTableViewCellId];
        return cell;
    }
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    CGFloat cellHeight = 54.0f;
    if (indexPath.row == infoModels.count - 1) {
        return 80.0f;
    }
    return cellHeight;
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/InfoScreenViewController/InfoTableViewController.swift
//
//  InfoTableViewController.swift
//  sample-conference-videochat-swift
//
//  Created by Injoit on 1/28/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

import UIKit
import Quickblox
import QuickbloxWebRTC

struct InfoCellConstant {
    static let infoTableViewCellId = "InfoTableViewCell"
    static let logoTableViewCellId = "QBLogoTableViewCell"
}

class InfoTableViewController: UITableViewController {
    
    //MARK: - Properties
    var infoModels = [InfoModel]()
    
    //MARK: - Life Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let backButtonItem = UIBarButtonItem(image: UIImage(named: "chevron"),
                                                           style: .plain,
                                                           target: self,
                                                           action: #selector(didTapBack(_:)))
        navigationItem.leftBarButtonItem = backButtonItem
        backButtonItem.tintColor = .white
        
        setupTableView()
    }
    
    @objc func didTapBack(_ sender: UIBarButtonItem) {
        navigationController?.popViewController(animated: true)
    }
    
    //MARK: - Internal Methods
    private func setupTableView() {
        let appName = Bundle.main.infoDictionary?["CFBundleName"] as? String
        let appNameModel = InfoModel()
        appNameModel.title = "Application name"
        appNameModel.info = appName ?? ""
        infoModels.append(appNameModel)
        
        let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
        let appVersionModel = InfoModel()
        appVersionModel.title = "Application version"
        appVersionModel.info = appVersion ?? ""
        infoModels.append(appVersionModel)
        
        let quickBloxSdkVersionModel = InfoModel()
        quickBloxSdkVersionModel.title = "QuickBlox SDK version"
        quickBloxSdkVersionModel.info = QuickbloxFrameworkVersion
        infoModels.append(quickBloxSdkVersionModel)
        
        let wbrtcVersionModel = InfoModel()
        wbrtcVersionModel.title = "QuickbloxWebRTC version:"
        wbrtcVersionModel.info = QuickbloxWebRTCFrameworkVersion
        infoModels.append(wbrtcVersionModel)
        
        let appIDModel = InfoModel()
        appIDModel.title = "Application ID"
        appIDModel.info = "\(QBSettings.applicationID)"
        infoModels.append(appIDModel)
        
        let authKeyModel = InfoModel()
        authKeyModel.title = "Auhtorization key"
        authKeyModel.info = QBSettings.authKey ?? ""
        infoModels.append(authKeyModel)
        
        let authSecretModel = InfoModel()
        authSecretModel.title = "Auhtorization secret"
        authSecretModel.info = QBSettings.authSecret ?? ""
        infoModels.append(authSecretModel)
        
        let accountKeyModel = InfoModel()
        accountKeyModel.title = "Account key"
        accountKeyModel.info = QBSettings.accountKey ?? ""
        infoModels.append(accountKeyModel)
        
        let apiDomainModel = InfoModel()
        apiDomainModel.title = "API domain"
        apiDomainModel.info = QBSettings.apiEndpoint ?? ""
        infoModels.append(apiDomainModel)
        
        let chatDomainModel = InfoModel()
        chatDomainModel.title = "Chat domain"
        chatDomainModel.info = QBSettings.chatEndpoint ?? ""
        infoModels.append(chatDomainModel)
        
        let janusServerURLModel = InfoModel()
        janusServerURLModel.title = "Janus server url:"
        janusServerURLModel.info = "\(QBRTCConfig.conferenceEndpoint() ?? "")"
        infoModels.append(janusServerURLModel)
        
        let qaVersion = Bundle.main.infoDictionary?["CFBundleVersion"] as? String
        let qaVersionModel = InfoModel()
        qaVersionModel.title = "QA version"
        qaVersionModel.info = qaVersion ?? ""
        infoModels.append(qaVersionModel)
        
        let logoModel = InfoModel()
        logoModel.title = "logo"
        logoModel.info = "logo"
        infoModels.append(logoModel)
        
        tableView.reloadData()
    }
    
    // MARK: - Table view data source
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return infoModels.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let isLast = indexPath.row == infoModels.count - 1
        if isLast == true {
            let cell = tableView.dequeueReusableCell(withIdentifier: InfoCellConstant.logoTableViewCellId, for: indexPath)
            return cell
        }
        guard let cell = tableView.dequeueReusableCell(withIdentifier: InfoCellConstant.infoTableViewCellId)
            as? InfoTableViewCell else {
                return UITableViewCell()
        }
        let model = infoModels[indexPath.row]
        cell.applyInfo(model: model)
        return cell
    }
    
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        let isLast = indexPath.row == infoModels.count - 1
        return isLast ? 80.0 : 54.0
    }
}
